File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
import sys
import math
input = sys.stdin.readline


while 1:
    n = int(input())
    cnt = 0
    if n==0:
        break
    for i in range(n+1, 2*n+1):
        if i==1:
            continue
        for j in range(2, int(math.sqrt(i)) + 1):
            if i%j==0:
                break
        else:
            cnt += 1
    print(cnt)

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N, count;
        while(true){
            N = Integer.parseInt(br.readLine());
            if(N==0)break;

            count = 0;
            for (int i = N+1; i <= 2*N; i++) {
                if(isPrime(i))count++;
            }
            bw.write(count + "\n");
        }

        bw.flush();
        bw.close();
        br.close();
    }

    private static boolean isPrime(int num) {
        if(num<2)return false;
        if(num==2)return true;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if(num%i==0)return false;
        }
        return true;
    }

}
